mantidproject/mantid
# Indirect - Vesuvio - Corrections for back-scattering spectra

Currently, corrections for back-scattering spectra (done indirectly through a vesuvio user script or directly using the VesuvioCorrections algorithm) incorrectly perform Multiple-scattering corrections and Gamma corrections (should never be done for back-scattering spectra). Multiple scattering corrections for a hydrogenous sample, across the back-scattering spectra, do not take into account the impact of hydrogen in the sample, this is due to the hydrogen peak being outside of the measurable TOF range.

In order to account for hydrogen (when it is present in the sample), an approximated intensity for the hydrogen peak (which lies outside the measured range) is calculated, by first providing users a method of constraining the quantity of hydrogen in the sample by one or more other known masses. These constraints can be used to approximate the intensity of the hydrogen peak using:
![ves_eq_mult](https://user-images.githubusercontent.com/15521164/29977399-1a9ff264-8f35-11e7-8e04-a4c3b1d2f363.PNG)
The above is in the case of multiple constraints, this can be simplified to:
![ves_eq_one](https://user-images.githubusercontent.com/15521164/29974741-d53edf40-8f2b-11e7-8829-8505c6fd1871.PNG)
In the case of a single constraint.

N is the number of constraints
M<sub>i</sub> is the mass used for the i<sup>th</sup> constraint
σ<sub>M</sub> is the cross-section of the mass M
I<sub>M</sub> is the intensity of mass M
α<sub>i</sub> is the weight of the i<sup>th</sup> constraint
β<sub>i</sub> is the constraint factor defined in the i<sup>th</sup> constraint

**To test:**
1. Load the supplied, original script (*_original.py) into the Mantid script editor.
2. Execute the script in Mantid and wait for the algorithm to finish executing.
3. Check the output contains the following: 
    - A time of flight (\*_tof) workspace
    - Two correction scale workspaces (\*\_correction\_fit\_scale\_iteration\__NUM_); one for each iteration
    - Two fitting data group workspaces (\*\_iteration\__NUM_\_data); one for each iteration
        - Should contain a Workspace2D for each spectrum (3-5)
    - Two correction workspaces (\*\_iteration\__NUM_); one for each iteration
        - Should contain the following correction workspaces:
            - TotalScattering (\*\_correction_TotalScattering)
            - GammaBackground (\*\_correction_GammaBackground)
            - MultipleScattering (\*\_correction_MultipleScattering)
        - Should contain the following corrected workspaces:
            - TotalScattering (\*\_corrected_TotalScattering)
            - GammaBackground (\*\_corrected_GammaBackground)
            - MultipleScattering (\*\_corrected_MultipleScattering)
    - Two parameter workspaces (\*\_params_iteration\__NUM_); one for each iteration
    - Two pre-correction parameter workspaces (\*\_params_pre_correction_iteration\__NUM_); one for each iteration
4. Load the supplied, modified script (\*\_modified.py) into the Mantid script editor.
5. Execute the script in Mantid and wait for the algorithm to finish executing.
6. Check the output contains the following:
    - A time of flight (\*_tof) workspace
    - Two correction scale workspaces (\*\_correction\_fit\_scale\_iteration\__NUM_); one for each iteration
    - Two fitting data group workspaces (\*\_iteration\__NUM_\_data); one for each iteration
        - Should contain a Workspace2D for each spectrum (135-140)
    - Two correction workspaces (\*\_iteration\__NUM_); one for each iteration
        - Should contain the following correction workspaces:
            - TotalScattering (\*\_correction_TotalScattering)
            - MultipleScattering (\*\_correction_MultipleScattering)
        - Should contain the following corrected workspaces:
            - TotalScattering (\*\_corrected_TotalScattering)
            - MultipleScattering (\*\_corrected_MultipleScattering)
    - Two parameter workspaces (\*\_params_iteration\__NUM_); one for each iteration
    - Two pre-correction parameter workspaces (\*\_params_pre_correction_iteration\__NUM_); one for each iteration

**Scripts**: 
[user_script_simplified_output.zip](https://github.com/mantidproject/mantid/files/1274506/user_script_simplified_output.zip)



Fixes #20301  

**Release Notes** 
Changes have been added to the release notes.


---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
